feature_generation/feature_generation.py

```diff
@@ -0,0 +1,39 @@
+import os
+from PIL import Image
+import numpy as np
+
+def get_training_data(training_data_directory):
```

Comment:
On a high level, I'd like to start refactoring this project to operate more like a library - the end result will be something with a nice API that can be integrated into a larger codebase. 

To this end, let's organize all of the functionality that is just for making your life easier into a separate folder (let's call it "workbench" or similar). You're welcome to organize _that_ folder as well (e.g., this function could go into an "io" package). Up2you.